Add escriptize provider

pguyot opened the pull request:
Requires support for AtomVM:
https://github.com/atomvm/AtomVM/pull/1948

NB: it's currently incompatible with JIT

pguyot committed:
Add escriptize provider

Requires support for AtomVM:
https://github.com/atomvm/AtomVM/pull/1948

NB: it's currently incompatible with JIT

Signed-off-by: Paul Guyot <pguyot@kallisys.net>

UncleGrumpy commented on src/atomvm_escriptize_provider.erl:
```diff
@@ -0,0 +1,456 @@
+-define(PROVIDER, escriptize).
+-define(DEPS, [packbeam]).
+-define(OPTS, [
+    {atomvm_binary, $b, "atomvm_binary", string, "Path to AtomVM binary (default: which AtomVM)"},
```
if we default to `which atomvm`, and use the launcher script from an installed AtomVM then we don't need to have atomvmlib supplied either. For those not using an installed AtomVM they will need to supply both.

UncleGrumpy commented on src/atomvm_escriptize_provider.erl:
```diff
@@ -0,0 +1,456 @@
+-define(DEPS, [packbeam]).
+-define(OPTS, [
+    {atomvm_binary, $b, "atomvm_binary", string, "Path to AtomVM binary (default: which AtomVM)"},
+    {atomvmlib, $l, "atomvmlib", string, "Path to atomvmlib.avm"},
```
Lets also add `(default: auto-detected if AtomVM is installed)`

UncleGrumpy reviewed (changes requested):
I am still reviewing some of the more complex sections, but I have a few simple fixes and requests that I noticed in a first pass. It would also be good to have some tests to verify the assembled escript.

UncleGrumpy committed:
Small adjustments to escriptize provider and test (#1)

This makes a few small adjustments to the escriptize provider as well as the tests, most of these
are necessary changes. A few additional changes have heen made, such as adding the users
$HOME/.local, to the default AtomVV install search paths, and supporting matching on the AtomVM elf
executable, as well as the shell launcher script.

Signed-off-by: Winford <winford@object.stream>

pguyot committed:
Detect executable with file(1)

Signed-off-by: Paul Guyot <pguyot@kallisys.net>

UncleGrumpy reviewed (approved):
This looks good to me now, and the test passes.

pguyot committed:
Do not update flags of .atomvm_avm_info section

This will prevent objcopy from dammaging the binary on linux/arm64

Signed-off-by: Paul Guyot <pguyot@kallisys.net>
